src/cursor_on_target.py

```diff
@@ -27,31 +27,82 @@
 import uuid
 import time
 import socket
-#import struct
-import socket
 import logging
+import math
+import hashlib
+import argparse
+import config
+
+# NATO phonetic alphabet
+NATO_ALPHABET = [
+    "ALPHA", "BRAVO", "CHARLIE", "DELTA", "ECHO", "FOXTROT", "GOLF", "HOTEL", "INDIA",
+    "JULIET", "KILO", "LIMA", "MIKE", "NOVEMBER", "OSCAR", "PAPA", "QUEBEC", "ROMEO",
+    "SIERRA", "TANGO", "UNIFORM", "VICTOR", "WHISKEY", "XRAY", "YANKEE", "ZULU"
+]
 
-"""
-    Generate a unique identifier for CoT messages.
 
+def get_mac_address():
+    """
+    Retrieve the MAC address of the device.
+    
     Returns:
-    str: A unique identifier string prefixed with 'OpenAthena-'.
+    str: The MAC address in the format "XX:XX:XX:XX:XX:XX"
+    """
+    return ':'.join(['{:02x}'.format((uuid.getnode() >> elements) & 0xff) for elements in range(0,8*6,8)][::-1])
+
 
-    Note:
-    This function uses UUID4 to ensure uniqueness across multiple instances
-    and executions of the software.
-"""
 def create_uid():
-    return f"OpenAthena-{uuid.uuid4()}"
+    """
+    Create a unique device identifier based on the MAC address.
+    
+    Returns:
+    str: A device UID in the format "[NATO_ALPHABET][00-99]"
+    """
+    mac = get_mac_address()
+    hash_value = hashlib.md5(mac.encode()).hexdigest()
+    index = int(hash_value[:4], 16) % 2600
+    letter_index = index // 100
+    number = index % 100
+    return f"{NATO_ALPHABET[letter_index]}{number:02d}"
 
-"""
+DEVICE_UID = create_device_uid()
+CALCULATION_SERIAL = 0
```

Comment:
As written, the program will always use a serial number of 1 since the value is not saved to disk. This will cause previously sent points to always get overridden. Functionality should be added to save and restore the next serial number from a file